core/drivers/crypto/caam/include/caam_desc_defines.h

```diff
@@ -0,0 +1,658 @@
+#define PKHA_CPY_SRC_A2			(PKHA_REG_SRC(A) | PKHA_SEG_SRC(2))
+#define PKHA_CPY_SRC_A3			(PKHA_REG_SRC(A) | PKHA_SEG_SRC(3))
+#define PKHA_CPY_SRC_B0			(PKHA_REG_SRC(B) | PKHA_SEG_SRC(0))
+#define PKHA_CPY_SRC_B1			(PKHA_REG_SRC(B) | PKHA_SEG_SRC(1))
+#define PKHA_CPY_SRC_B2			(PKHA_REG_SRC(B) | PKHA_SEG_SRC(2))
+#define PKHA_CPY_SRC_B3			(PKHA_REG_SRC(B) | PKHA_SEG_SRC(3))
+#define PKHA_CPY_SRC_N0			(PKHA_REG_SRC(N) | PKHA_SEG_SRC(0))
+#define PKHA_CPY_SRC_N1			(PKHA_REG_SRC(N) | PKHA_SEG_SRC(1))
+#define PKHA_CPY_SRC_N2			(PKHA_REG_SRC(N) | PKHA_SEG_SRC(2))
+#define PKHA_CPY_SRC_N3			(PKHA_REG_SRC(N) | PKHA_SEG_SRC(3))
+
+#define PKHA_CPY_DST(dst)		PKHA_CPY_DST_##dst
+#define PKHA_CPY_DST_A0			(PKHA_REG_DST(A) | PKHA_SEG_DST(0))
+#define PKHA_CPY_DST_A1			(PKHA_REG_DST(A) | PKHA_SEG_DST(1))
+#define PKHA_CPY_DST_A2			(PKHA_REG_DST(A) | PKHA_SEG_DST(2))
+#define PKHA_CPY_DST_A3			(PKHA_REG_DST(A) | PKHA_SEG_DST(3))
+#define PKHA_CPY_DST_B0			(PKHA_REG_DST(B) | PKHA_SEG_DST(0))
+#define PKHA_CPY_DST_B1			(PKHA_REG_DST(B) | PKHA_SEG_DST(1))
+#define PKHA_CPY_DST_B2			(PKHA_REG_DST(B) | PKHA_SEG_DST(2))
+#define PKHA_CPY_DST_B3			(PKHA_REG_DST(B) | PKHA_SEG_DST(3))
+#define PKHA_CPY_DST_N0			(PKHA_REG_DST(N) | PKHA_SEG_DST(0))
+#define PKHA_CPY_DST_N1			(PKHA_REG_DST(N) | PKHA_SEG_DST(1))
+#define PKHA_CPY_DST_N2			(PKHA_REG_DST(N) | PKHA_SEG_DST(2))
+#define PKHA_CPY_DST_N3			(PKHA_REG_DST(N) | PKHA_SEG_DST(3))
+#define PKHA_CPY_DST_E			(PKHA_REG_DST(E))
+
+/*
+ * Descriptor Protocol Data Block
+ */
+/* RSA Encryption */
+#define PDB_RSA_ENC_E_SIZE(len)		SHIFT_U32((len & 0xFFF), 12)
+#define PDB_RSA_ENC_N_SIZE(len)		SHIFT_U32((len & 0xFFF), 0)
+#define PDB_RSA_ENC_F_SIZE(len)		SHIFT_U32((len & 0xFFF), 0)
+
+/* RSA Decryption */
+#define PDB_RSA_DEC_D_SIZE(len)		SHIFT_U32((len & 0xFFF), 12)
+#define PDB_RSA_DEC_N_SIZE(len)		SHIFT_U32((len & 0xFFF), 0)
+#define PDB_RSA_DEC_Q_SIZE(len)		SHIFT_U32((len & 0xFFF), 12)
+#define PDB_RSA_DEC_P_SIZE(len)		SHIFT_U32((len & 0xFFF), 0)
+
+/* RSA Finalize Key */
+#define PDB_RSA_KEY_P_SIZE(len)		SHIFT_U32((len & 0x1FF), 0)
+#define PDB_RSA_KEY_E_SIZE(len)		SHIFT_U32((len & 0x3FF), 0)
+#define PDB_RSA_KEY_N_SIZE(len)		SHIFT_U32((len & 0x3FF), 16)
+
+/* Manufacturing Curve Select */
+#define PDB_MP_CSEL_P256		0x03
+#define PDB_MP_CSEL_P384		0x04
+#define PDB_MP_CSEL_P521		0x05
+
+/* Public Key Generation */
+#define PDB_PKGEN_PD1				SHIFT_U32(1, 25)
+/* Public Key Signature */
+#define PDB_PKSIGN_PD1				SHIFT_U32(1, 22)
+/* Public Key Verify */
+#define PDB_PKVERIFY_PD1			SHIFT_U32(1, 22)
+/* Shared Secret */
+#define PDB_SHARED_SECRET_PD1		SHIFT_U32(1, 25)
+
+
+/* ECC Domain Selection */
+#define PDB_ECC_ECDSEL(curve)		SHIFT_U32((curve & 0x3F), 7)
+
+/* Black key padding */
+#define BLACK_KEY_NONCE_SIZE		6
+#define BLACK_KEY_ICV_SIZE			6
+
+/*
+ * ECC Predefined Domain
+ */
+enum caam_ecc_curve {
+	ECC_P192 = (0x00),
+	ECC_P224,
+	ECC_P256,
+	ECC_P384,
+	ECC_P521,
+	ECC_MAX,
+	ECC_UNKNOWN = (0xFF),
```

Comment:
I'll change the enumerate but as you said this only apply to the CAAM and file is not globally exported. I prefer to keep rest as is because it will increase macro name size that are not to be used outside the CAAM driver.